src/main/java/nl/tudelft/contextproject/model/level/MSTBasedLevelFactory.java

```diff
@@ -0,0 +1,332 @@
+	public ArrayList<RoomExitPoint> usedExitPoints;
+	public int idCounter;
+	public RoomTuple startAndEndRooms;
+	public ArrayList<Light> lights;
+	public MazeTile[][] mazeTiles;
+
+
+	/**
+	 * Constructor.
+	 * 
+	 * @param mapFolder
+	 * 		location and name of mapFolder
+	 */
+	public MSTBasedLevelFactory(String mapFolder) {
+		lights = new ArrayList<Light>();
+		mazeTiles = new MazeTile[MAX_WIDTH][MAX_HEIGHT];
+		initializeBuilder(mapFolder);
+	}
+
+	/**
+	 * Generate a level in 6 steps.
+	 *<ol>
+	 *	<li>Place start and treasure room.</li>
+	 *	<li>Place randomly rooms for some amount of attempts.</li>
+	 *	<li>Create an edge from every entrance door to every exit door of other rooms.</li>
+	 *	<li>Create MST. </li>
+	 *	<li>For every edge still existing use breadth first search to find corridors.</li>
+	 *	<li>Beautify corridors.</li>
+	 *	<li>Add all placed entities in the entity list.</li>
+	 *</ol>
+	 */
+	@Override
+	public Level generateSeeded(long seed) {
+		createRNG(seed);
+		placeStartAndTreasureRoom();
+		placeOtherRooms();
+		createEdges();
+		createReverseMST();
+		placeCorridors();
+		MSTBasedLevelFactory.carveCorridorWalls(mazeTiles);
+		for (RoomNode roomNode: usedNodes) {
+			Room room = roomNode.room;
+			entities.addAll(room.entities);
+		}
+		return new Level(mazeTiles, lights);
+	}
+
+	/**
+	 * Place start and treasure room on semi-random locations.
+	 * Start room is placed in left-most quarter, treasure-room in right most quarter.
+	 */
+	public void placeStartAndTreasureRoom() {
+		int endLeftMostQuarter = (int) Math.round(MAX_WIDTH / 4.0);
+		int beginningRightMostQuarter = (int) Math.round(3.0 * MAX_WIDTH / 4.0);
+
+		//Place start room
+		Vec2I startLocation = new Vec2I(RandomUtil.getRandomIntegerFromInterval(rand, 
+				RoomNode.MIN_DIST, endLeftMostQuarter), 
+				RandomUtil.getRandomIntegerFromInterval(rand, 
+						RoomNode.MIN_DIST, MAX_HEIGHT - (startAndEndRooms.getStarterRoom().size.getWidth() + RoomNode.MIN_DIST + 1)));
+		RoomNode startNode = new RoomNode(startAndEndRooms.getStarterRoom(), START_ROOM_ID);
+		addRoomNode(startNode, startLocation);
+
+		//Place treasure room
+		Vec2I treasureLocation = new Vec2I(RandomUtil.getRandomIntegerFromInterval(rand, 
+				beginningRightMostQuarter, MAX_WIDTH - (startAndEndRooms.getTreasureRoom().size.getWidth() + RoomNode.MIN_DIST + 1)), 
+				RandomUtil.getRandomIntegerFromInterval(rand, 
+						RoomNode.MIN_DIST, MAX_HEIGHT - (startAndEndRooms.getTreasureRoom().size.getHeight() + RoomNode.MIN_DIST + 1)));
+		RoomNode treasureNode = new RoomNode(startAndEndRooms.getTreasureRoom(), TREASURE_ROOM_ID);
+		addRoomNode(treasureNode, treasureLocation);
+	}
+
+	/**
+	 * Place the other rooms.
+	 * Will try to place random rooms until max attempts has been reached or there are no rooms left to place.
+	 */
+	public void placeOtherRooms() {
+		int attempts = 0;
+		int randomIndex;
+		RoomNode currentNode;
+		while (attempts <= MAX_ATTEMPTS && !baseNodes.isEmpty()) {
```

Comment:
Technically true. Good catch, though it does not matter a lot if you are trying a 100 to a 100 thousand times to try once more  xD.